rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -900,13 +892,63 @@ public boolean afterLRANotification() {
         return notifiedAll;
     }
 
-    static class AfterLRAListener {
-        public AfterLRAListener(URI notificationURI, URI recoveryId) {
+    private abstract static class EndLRAListener {
```

Comment:
Please can you explain a little of the motivation of the hierarchy of the EndLRAListener, AfterLRAListener and ForgetLRAListener. 

I am wondering about whether some further symmetry in the structure of the implementation of the call(Client) method could be achieved? As an example they appear to diverge slightly in the exception handling (typed exceptions vs Exception)?

Reply:
the JAX-RS calls they are making are different at least AfterLRA is PUT vs Forget is DELETE. There are also some differences in the passed headers - we don't pass LRA id to AfterLRA because a new LRA can be started on these methods but instead, we are passing LRA ENDED header. Also, the log messages in case of exceptions differ. I don't want to rule out that any further alignment can be made but for this PR I would prefer to just stick with refactoring.

Reply:
I see, thanks